src/raw/client.rs

```diff
@@ -89,6 +95,22 @@ impl Client {
         Client {
             rpc: self.rpc.clone(),
             cf: Some(cf),
+            atomic: self.atomic,
+        }
+    }
+
+    /// Set to use the atomic mode.
+    ///
+    /// The only reason of using atomic mode is the
+    /// [`compare_and_swap`](Client::compare_and_swap) operation. To guarantee
+    /// the atomicity of CAS, write operations like [`put`](Client::put) or
+    /// [`delete`](Client::delete) in atomic mode are more expensive. Some
+    /// operations are not supported in the mode.
+    pub fn with_atomic(&self) -> Client {
```

Comment:
maybe we should rename to `with_atomic_for_cas` so as not to give the impression that the user should use this to get atomic put/delete (which are atomic in the usual sense with or without this option).

Reply:
Yeah it makes a lot of sense